File: 3864-count-the-number-of-computer-unlocking-permutations/3864-count-the-number-of-computer-unlocking-permutations.java
class Solution {
    public int countPermutations(int[] complexity) {
        int n = complexity.length;
        long ans= 1 ;
        int MOD = 1_000_000_007;
        for(int i = 1 ; i < n ; i++){
            if(complexity[i]<=complexity[0]) return 0;
            ans = (ans*i)%MOD;
        }
        return (int)ans;
    }
}

File: 3864-count-the-number-of-computer-unlocking-permutations/3864-count-the-number-of-computer-unlocking-permutations.cpp
class Solution {
public:
    int countPermutations(vector<int>& complexity) {
        long long mod = 1000000007;
        long long ans = 1;
        for(int i = 1; i < complexity.size() ; i++){
            if(complexity[i] <= complexity[0]) return 0;
            ans = (ans*i)%mod;       
        }
            return (int)ans;
    }
};